packages/amplify_core/lib/src/types/exception/api/http_status_exception.dart

```diff
@@ -3,19 +3,22 @@
 
 import 'package:amplify_core/amplify_core.dart';
 
-/// {@template rest_exception}
+@Deprecated('Use HttpStatusException instead')
+typedef RestException = HttpStatusException;
+
+/// {@template amplify_core.api.http_status_exception}
 /// An HTTP error encountered during a REST API call, i.e. for calls returning
 /// non-2xx status codes.
 /// {@endtemplate}
-class RestException extends ApiException {
+class HttpStatusException extends ApiException {
   /// {@macro rest_exception}
-  RestException(this.response) : super(response.decodeBody());
+  HttpStatusException(this.response) : super(response.decodeBody());
 
   /// The HTTP response from the server.
   final AWSHttpResponse response;
 
   @override
   String toString() {
-    return 'RestException{response=$response}';
+    return 'HttpStatusException{response=$response}';
   }
```

Comment:
I would think we'd want to print the `response.decodeBody()` instead which is effectively `this.message`, which is already printed without overriding `toString`. I would remove this override and just update `runtimeTypeName` instead.

Reply:
`response.statusCode` should also be printed, I would think, given it's a status exception